init-common/functions

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+qsvc() {
+    # Returns whether a service is enabled.
+    # Usage: qsvc <nameofservice>
+    # Must only be used after qubes-sysinit has started.
+    test -e /var/run/qubes-service/"$1"
+}
+
+under_systemd() {
+    pidof systemd >/dev/null 2>&1
+}
+
+possibly_run_save_script()
+{
+    ENCODED_SCRIPT=$(qubesdb-read /qubes-save-script)
+    if [ -z "$ENCODED_SCRIPT" ] ; then return ; fi
+    tmpfile=$( mktemp qubes-save-script.XXXXXXXXX )
+    echo $ENCODED_SCRIPT|perl -e 'use MIME::Base64 qw(decode_base64); local($/) = undef;print decode_base64(<STDIN>)' >"$tmpfile"
+    chmod 755 "$tmpfile"
+    Xorg -config /etc/X11/xorg-preload-apps.conf :0 &
+    while ! [ -S /tmp/.X11-unix/X0 ]; do sleep 0.5; done
+    DISPLAY=:0 su - user -c "$tmpfile"
+    ret=$?
+    killall Xorg
+    rm -f "$tmpfile"
+    return $ret
+}
```

Comment:
I'm not sure if `Wants=` dependency is a good idea. It will pull in `qubes-gui-agent.service` even when qubes-dvm start condition isn't met. So it will make much harder to create GUI-less VM. Anyway, it isn't easy in Qubes 3.x, will be much easier in Qubes 4.x. So we may fix it there, when there will be a need.

As for ordering in sysvinit - previously it was intended to start as early as /rw got mounted, before gui-agent as it started own X server. But now it should be started as late as possible. The less things needs to be done after, the faster DispVM startup will be.
Separate initscript is fine.